.github/pull_request_template.md

```diff
@@ -0,0 +1,7 @@
+Enhancement:
+
+Reason:
+
+Result:
+
+Issue Tracker Tickets (Jira or BZ if any):
```

Comment:
> This template matches only for the enhancement PRs. There are bugfixes and other options. Not even your PR follows this template so I find it kind of artificial.
> 
 It is possible to add different templates, e.g. have different fields requested for fix and for feat, but that's pretty ugly, described in https://stackoverflow.com/questions/73771068/multiple-templates-for-pull-requests-on-github
So users must do couple clicks to navigate to their template, it's not possible to set the template based on the PR title prefix. So it is easier to just have one template and then rely on contributors' common sense to decide how to fill in the template.
> In the templates, I would propose to have some generic checklists for contributor/reviewer se have for example in libssh (example: https://gitlab.com/libssh/libssh-mirror/-/merge_requests/340). We probably do not need all the steps, but I find some of them usefule, at least for example the following
> 
>     * [ ]  Documentation is updated
> 
>     * [ ]  Code is modified for a feature
> 
>     * [ ]  Tests are implemented/updated

I think that would be an overkill to impose so much control, not all commits require this. For now, the aim is to make pr description to have enough information to eventually put it into `doc text` field in BZ or Jira and have it reviewed by writers without the need to ping developers who worked on the subject several months ago.
And of course get PR title and description to changelog.

Reply:
> > This template matches only for the enhancement PRs. There are bugfixes and other options. Not even your PR follows this template so I find it kind of artificial.
> 
> It is possible to add different templates, e.g. have different fields requested for fix and for feat, but that's pretty ugly, described in https://stackoverflow.com/questions/73771068/multiple-templates-for-pull-requests-on-github So users must do couple clicks to navigate to their template, it's not possible to set the template based on the PR title prefix. So it is easier to just have one template and then rely on contributors' common sense to decide how to fill in the template.

It would help if the template stated that - maybe some text like
```
# This template is for enhancements because there isn't an easy way to have different templates automatically
# selected for different PR types.  Therefore, use your best judgement.  Note that for features and bug fixes, it is
# very useful for end users to have a detailed description.
```
Then the tooling could strip out the lines matching `^#`

> 
> > In the templates, I would propose to have some generic checklists for contributor/reviewer se have for example in libssh (example: https://gitlab.com/libssh/libssh-mirror/-/merge_requests/340). We probably do not need all the steps, but I find some of them usefule, at least for example the following
> > ```
> > * [ ]  Documentation is updated
> > 
> > * [ ]  Code is modified for a feature
> > 
> > * [ ]  Tests are implemented/updated
> > ```
> 
> I think that would be an overkill to impose so much control, not all commits require this. For now, the aim is to make pr description to have enough information to eventually put it into `doc text` field in BZ or Jira and have it reviewed by writers without the need to ping developers who worked on the subject several months ago. And of course get PR title and description to changelog.

I think having a checklist is a separate issue, and a good idea - I just recently had to go back and request a kdump developer to add tests to a PR because I missed them - maybe have a 4th checklist item like
```
* [] None of the above - not applicable for this PR
```
or something like that - for tests only PRs, ci, docs, etc.

Reply:
The template supports comments using HTML comments like `<!-- commment -->` so the instructions can be written like this.

We have similar checklist in opensc with comments: https://github.com/OpenSC/OpenSC/blob/master/.github/PULL_REQUEST_TEMPLATE.md?plain=1 with instructions to remove non-applicable items so that would be also an option: `<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->`

Reply:
The html comments are preserved in the pr body e.g.
```
        "body": "fixes https://github.com/OpenSC/OpenSC/issues/2444\r\n\r\n<!--\r\nThank you for your pull request.\r\n\r\nIf this fixes a GitHub issue, make sure to have a line saying 'Fixes #XXXX'\r\n(without quotes) in the commit message.\r\n\r\nMention which card(s) are used during testing. To get the name of your card,\r\nrun this command: `opensc-tool -n`\r\n-->\r\n\r\n##### Checklist\r\n<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->\r\n- [ ] Documentation is added or updated\r\n- [ ] New files have a LGPL 2.1 license statement\r\n- [ ] PKCS#11 module is tested\r\n- [ ] Windows minidriver is tested\r\n- [ ] macOS tokend is tested\r\n",
        "mergedAt": "2023-02-27T10:39:08Z",
        "number": 2501,
        "title": "enable use_file_cache by default"
```
the html style comments would make automated tooling difficult - I would prefer to use lines beginning with `^#` as comments to make it easy for the automated tooling to strip out